### PR TITLE
Support for setting context, dockerfile, target and build args in "ok…

### PR DESCRIPTION
**Pull request by pchico83**
…teto push"

Signed-off-by: Pablo Chico de Guzman <pchico83@gmail.com>

Fixes https://github.com/okteto/okteto/issues/700#issuecomment-617260201

### CODE REVIEW

File: pkg/model/dev.go [hunk 1]
```diff
@@ -91,6 +91,7 @@ type Dev struct {
 	Namespace            string                `json:"namespace,omitempty" yaml:"namespace,omitempty"`
 	Container            string                `json:"container,omitempty" yaml:"container,omitempty"`
 	Image                string                `json:"image,omitempty" yaml:"image,omitempty"`
+	Push                 *BuildInfo            `json:"-,omitempty" yaml:"push,omitempty"`
```
Comment (rberrelleza) [inline pkg/model/dev.go:94]: Yes!!!!

should we call it `build` since it's buildinfo?

File: pkg/model/dev.go [hunk 2]
```diff
@@ -111,6 +112,14 @@ type Dev struct {
 	SSHServerPort        int                   `json:"sshServerPort,omitempty" yaml:"sshServerPort,omitempty"`
 }
 
+// BuildInfo represents the build info to generate an image
+type BuildInfo struct {
```
Comment (rberrelleza) [inline pkg/model/dev.go:116]: Will this be overridable via the CLI?

File: pkg/model/dev.go (hunk 1 above)
Comment (rberrelleza) [reply pkg/model/dev.go:94]: and could we also use it in `build`?

Comment (pchico83) [reply pkg/model/dev.go:94]: I talked to @rlamana  and the idea is to have a `build` field for building the dev image. `okteto up --build` would build your dev image on the fly. This is why we chose `push` here.

File: pkg/model/dev.go (hunk 2 above)
Comment (pchico83) [reply pkg/model/dev.go:116]: I can add it as part of this PR

File: pkg/model/dev.go (hunk 1 above)
Comment (rberrelleza) [reply pkg/model/dev.go:94]: cool

File: pkg/model/dev.go (hunk 2 above)
Comment (rberrelleza) [reply pkg/model/dev.go:116]: only if we have a customer ask, if not, let's keep it simple and wait?

Comment (pchico83) [reply pkg/model/dev.go:116]: Yeap, that is why I didn't add them.
Also, those parameters shouldn't change in the okteto workflow

**Pull request merged**